### ISSUE TITLE
pstats fails to start when ipv6 is disabled

### ISSUE THREAD

**Issue by Moguri** [panda3d/panda3d#558]
When disabling ipv6 with the `ipv6.disable=1` Linux kernel parameters, `pstats` fails to start. The two symptoms of this are:
* failing to open a socket/port (observed with 1.11 non-manylinux1 wheel)
* segfaulting (observed with 1.10.1 manylinux1 wheel)

We could try to work around this, but it may not be worth the effort. I propose we add a warning to the manual. I suspect most people running with ipv6 disabled will know they have it disabled and react to the warning in the manual.

Setting the `support-ipv6` PRC variable to `false` fixes this as does re-enabling ipv6 support in the kernel.

**Comment by CFSworks** [comment]
Panda's IPv6 support could use some enhancements either way. It should probably have a function that returns an enum indicating the level of confidence for IPv6 capability based on assigned network addresses and whether the kernel will provide an INET6 socket. It should probably also fall back from IPv6 to IPv4 (or vice versa depending on said enum) for failed TCP connections to a hostname with both address types.

**Comment by rdb** [comment]
Yeah, the IPv6 support I added was a minimal-effort approach to support IPv6 at all.  It needs to be more comprehensive.

In particular, `getaddrinfo` returns a list of results that we should try in sequence, but we only use the first one.  If the first result happens to be an IPv6 address (for some reason—it sounds like a Linux bug if an `AF_INET6` address is returned from `getaddrinfo()` while IPv6 is disabled), and it doesn't work, then Panda should fall back to the second and third results.  We will probably need to move the address resolution out of `Socket_Address::set_host()` to make this work.

Panda3D certainly shouldn't crash if it fails to open a connection.
